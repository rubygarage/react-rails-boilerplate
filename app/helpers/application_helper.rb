module ApplicationHelper
  def destroy_admin_user_session_path
    "/log_out"
  end
end
