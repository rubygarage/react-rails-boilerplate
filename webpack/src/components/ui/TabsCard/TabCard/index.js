import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Icon from 'components/ui/Icon'

import styles from './tab-card.css'

export class TabCard extends Component {
  static propTypes = {
    className: PropTypes.string,
    text: PropTypes.string,
    type: PropTypes.string,
    icon: PropTypes.string,
    color: PropTypes.string,
    iconClass: PropTypes.string,
    onClick: PropTypes.func,
    tabName: PropTypes.string
  }

  handleTabSwitch = () => {
    const { tabName } = this.props
    if (tabName === 'text') { // TODO remove this when text card will be implemented
      return
    }

    this.props.onClick(tabName)
  }

  render() {
    const { className, color, text, type, icon, iconClass, tabName } = this.props

    return (
      <div
        className={
          classNames(
            { [styles.disable]: tabName === 'text' }, // TODO remove this when text card will be implemented
            styles.tab,
            styles[type],
            styles[color],
            className,
            'tab-theme'
          )
        }
        onClick={this.handleTabSwitch}
      >
        <Icon type={icon} className={iconClass} />
        <span>
          {text}
        </span>
      </div>
    )
  }
}

export default TabCard
