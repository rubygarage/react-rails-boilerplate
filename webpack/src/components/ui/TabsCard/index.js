import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import styles from './tabs-card.css'

function TabsCard({ className, children }) {
  return (
    <div className={classNames(styles.tabs, className)}>
      {children}
    </div>
  )
}

TabsCard.propTypes = {
  className: PropTypes.string,
  children: PropTypes.array
}

export default TabsCard
