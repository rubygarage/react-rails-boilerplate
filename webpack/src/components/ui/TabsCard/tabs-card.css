.tabs {
  display: flex;
  justify-content: space-between;
  flex-shrink: 0;
}
