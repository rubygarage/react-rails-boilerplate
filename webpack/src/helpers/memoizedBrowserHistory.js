import createHistory from 'history/createBrowserHistory';

let instance = null;

export default class memoizedBrowserHistory {
  constructor() {
    if (!instance) {
      instance = createHistory();
    }

    return instance;
  }
}
