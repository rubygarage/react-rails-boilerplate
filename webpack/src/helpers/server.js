// moved the ugly environment (server/client) check to this helper
export const isServer = () => typeof window === 'undefined'
export const isClient = () => typeof window !== 'undefined'
