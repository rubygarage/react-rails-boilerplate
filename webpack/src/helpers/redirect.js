import memoizedBrowserHistory from 'helpers/memoizedBrowserHistory'

// uses browser router or server router depending on presence of response object in arguments
export const redirect = (path, res = null) => {
  if (res) {
    res.redirect(path)
    // we need to mark the response as finished to avoid double-render/double headers being set later in the stack
    // ( see src/server.js )
    res.finished = true
  } else {
    new memoizedBrowserHistory().push(path)
  }
}
