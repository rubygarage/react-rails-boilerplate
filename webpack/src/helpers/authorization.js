import { normalize } from 'normalize-json-api';
import { setTokenToStorage, deleteTokenFromStorage } from 'utils/tokens';
import { isClient } from 'helpers/server';
import ApiClient from 'utils/apiClient';

const authorizeUser = (req, resolve) => {
  const apiClient = new ApiClient().buildClient(req);

  apiClient.get('/api/v1/auth/users/session').then((response) => {
    const { entities, results } = normalize(response.data);

    if (isClient()) { setTokenToStorage(response.headers); }
    resolve({ currentUser: results, entities });
  }).catch(() => {
    deleteTokenFromStorage();
    resolve({ currentUser: {}, entities: {} });
  });
};

export default authorizeUser;
