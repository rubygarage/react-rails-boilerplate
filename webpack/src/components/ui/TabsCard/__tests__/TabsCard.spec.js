import React from 'react'
import { shallowWithIntl } from 'helpers/intl-enzyme'
import TabsCard from 'components/ui/TabsCard'

describe('<TabsCard />', () => {
  const props = {
    className: 'mb-15',
    children: []
  }

  it('renders tabs wraper', () => {
    const component = shallowWithIntl(
      <TabsCard {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
