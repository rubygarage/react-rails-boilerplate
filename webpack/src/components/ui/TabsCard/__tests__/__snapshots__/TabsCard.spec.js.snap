// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabsCard /> renders tabs wraper 1`] = `
<div
  className="tabs mb-15"
/>
`;
